src/java/htsjdk/samtools/cram/ref/ReferenceSource.java

```diff
@@ -40,14 +40,14 @@
 import java.util.Map;
 import java.util.regex.Pattern;
 
-public class ReferenceSource {
+public class ReferenceSource implements CRAMReferenceSource {
```

Comment:
Argh...in a just world `CRAMReferenceSource` would implement `ReferenceSource`, not vice versa. Can we rename `ReferenceSource` to something like `CRAMReferenceSourceImpl` to avoid confusion?